feat(azure): add configuration for Azure OpenAI Service (#1)
Unique-AG/openai-node: README.md

# Azure OpenAI NodeJS Library

This is a fork of the official [OpenAI Node.js library](https://github.com/openai/openai-node) that has been adapted to support the Azure OpenAI API. The objective of this library is to minimize the changes required to migrate from the official OpenAI library to Azure OpenAI or revert back to OpenAI.

> **Note**
> It is also an inofficial fork of [1openwindow/azure-openai-node](https://github.com/1openwindow/azure-openai-node) which we could not use due to due diligence reasons 🥇 

This library allows you to use Azure OpenAI's API without making any changes to your existing OpenAI code. You can simply add Azure information to your configuration and start using the Azure OpenAI model.

## Installation

```bash

> $ npm install @unique/openai

Reviewer: We need to see if that namespace is available before merging.

```

## Usage

> The library needs to be configured with your Azure OpenAI's key, endpoint and deploymentId. you can obtain these credentials from [Azure Portal](https://portal.azure.com).

Reviewer: ```suggestion
The library must be configured with your Azure OpenAI's `key`, `endpoint`, and `deploymentId`. You can obtain these credentials from [Azure Portal](https://portal.azure.com).
```

```ts

> import { Configuration, OpenAIApi } from '@unique/openai';

Reviewer: see above: might be another namespace.

const configuration = new Configuration({
  apiKey: process.env.OPENAI_API_KEY,
});

const openai = new OpenAIApi(
  new Configuration({
    apiKey: this.apiKey,
    // add azure info into configuration
    azure: {
      apiKey: <key1>
      endpoint: <endpoint>,

>     // deploymentName is optional, if you donot set it, you need to set it in the request parameter

Reviewer: ```suggestion
    // The `deploymentName` parameter is optional; if you do not set it, you need to put it in the request parameter
```
